aiohttp/client.py

```diff
@@ -304,6 +311,12 @@ def request(
         """Perform HTTP request."""
         return _RequestContextManager(self._request(method, url, **kwargs))
 
+    def _build_url(self, str_or_url: StrOrURL) -> URL:
+        if self._base_url is None:
+            return URL(str_or_url)
+        else:
+            return self._base_url.join(URL(str_or_url))
```

Comment:
```suggestion
    def _build_url(self, str_or_url: StrOrURL) -> URL:
        url = URL(str_or_url)
        if self._base_url is None:
            return url
        else:
            assert not url.is_absolute() and url.path.startswith('/')
            return self._base_url.join(url)
```
Let's restrict the code to conforming provided docs examples and avoiding edge cases.

Reply:
We can extend the support later